pandas/tests/frame/test_axis_select_reindex.py

```diff
@@ -117,6 +117,14 @@ def test_drop(self):
         df.drop(labels=df[df.b > 0].index, inplace=True)
         assert_frame_equal(df, expected)
 
+        # Empty DatetimeIndex/list, GH 27994
+        df_index = pd.DataFrame(
+            {"a": [0, 1]},
+            index=[pd.Timestamp("2019-01-01"), pd.Timestamp("2019-01-01")],
+        )
+        assert_frame_equal(df_index.drop([]), df_index)
+        assert_frame_equal(df_index.drop(df_index[df_index["a"] > 2].index), df_index)
```

Comment:
the expression here would be simpler with just `df_index.index[:0]`.  any reason not to use that?

`df_index.drop([])` should go on the previous line as `result = ...`.  ditto for the next assertion.

Reply:
As mentioned in https://github.com/pandas-dev/pandas/issues/27994#issuecomment-523471699, `TypeError` was raised when I tried `df.drop([])`. So I thought I should do two checks. But good thinking on `df_index.index[:0]`.